go/v1/statement_test.go

```diff
@@ -84,7 +84,20 @@ func TestBadStatementSubject(t *testing.T) {
 
 	err = got.Validate()
 
-	assert.Error(t, err, "Error: created malformed Statement (bad subject)")
+	// With this input, we encounter the RD object's validation error
+	assert.ErrorIs(t, err, ErrRDRequiredField, "Error: created malformed Statement (bad subject)")
```

Comment:
optional: I'm wondering if these should be separate tests?  Alternatively maybe this should be a table driven test? https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

Reply:
Updated to use table, I was trying to keep the diff small but this is def better.